python/ucxx/_lib_async/endpoint.py

```diff
@@ -85,6 +85,43 @@ async def close(self):
                 await asyncio.sleep(0)
                 self.abort()
 
+    async def am_send(self, buffer):
+        """Send `buffer` to connected peer via active messages.
+
+        Parameters
+        ----------
+        buffer: exposing the buffer protocol or array/cuda interface
+            The buffer to send. Raise ValueError if buffer is smaller
+            than nbytes.
+        """
+        self._ep.raise_on_error()
+        if self.closed():
+            raise UCXCloseError("Endpoint closed")
+        if not isinstance(buffer, Array):
+            buffer = Array(buffer)
+
+        # Optimization to eliminate producing logger string overhead
+        if logger.isEnabledFor(logging.DEBUG):
+            nbytes = buffer.nbytes
+            log = "[AM Send #%03d] ep: %s, nbytes: %d, type: %s" % (
+                self._send_count,
+                hex(self.uid),
+                nbytes,
+                type(buffer.obj),
+            )
+            logger.debug(log)
```

Comment:
This can be a little simpler since the logging module does at least accept format strings and arguments in its calls.

So

```suggestion
            logger.debug(
                "[AM Send #%03d] ep: 0x%x, nbytes: %d, type: %s", 
                self._send_count,
                self.uid,
                nbytes,
                type(buffer.obj),
            )
```

Reply:
Done in https://github.com/rapidsai/ucxx/pull/48/commits/bd1787f0b17d505f6e4ec60037fc605d057bcd01